frontend/app/src/store/history/index.ts

```diff
@@ -1,17 +1,937 @@
+    } catch (e: any) {
+      notify(e);
+    }
+  };
+
+  const ignoreInAccounting = async (
+    { actionIds, type }: IgnoreActionPayload,
+    ignore: boolean
+  ) => {
+    try {
+      ignore
+        ? await api.ignoreActions(actionIds, type)
+        : await api.unignoreActions(actionIds, type);
+      await fetchIgnored();
+    } catch (e: any) {
+      let title: string;
+      let description: string;
+      if (ignore) {
+        title = i18n.t('actions.ignore.error.title').toString();
+      } else {
+        title = i18n.t('actions.unignore.error.title').toString();
+      }
+
+      if (ignore) {
+        description = i18n
+          .t('actions.ignore.error.description', { error: e.message })
+          .toString();
+      } else {
+        description = i18n
+          .t('actions.unignore.error.description', { error: e.message })
+          .toString();
+      }
+      store.commit(
+        'setMessage',
+        {
+          success: false,
+          title,
+          description
+        } as Message,
+        { root: true }
+      );
+      return { success: false };
+    }
+
+    return { success: true };
+  };
+
+  // Purge
+  const purgeHistoryLocation = async (exchange: SupportedExchange) => {
+    useTrade().fetchTrades(true, exchange).then();
+    useAssetMovement().fetchAssetMovements(true, exchange).then();
+    useLedgerAction().fetchLedgerActions(true, exchange).then();
+  };
+
+  const purgeExchange = async (
+    exchange: SupportedExchange | typeof ALL_CENTRALIZED_EXCHANGES
+  ) => {
+    function resetStatus(section: Section) {
+      setStatus(Status.NONE, section, store.getters['status'], store.commit);
+    }
+    if (exchange === ALL_CENTRALIZED_EXCHANGES) {
+      resetStatus(Section.TRADES);
+      resetStatus(Section.ASSET_MOVEMENT);
+      resetStatus(Section.LEDGER_ACTIONS);
+    } else {
+      await purgeHistoryLocation(exchange);
+    }
+  };
+
+  // Gitcoin
+  const fetchGitcoinGrant = async (
+    payload: GitcoinGrantEventsPayload
+  ): Promise<ActionResult<GitcoinGrants>> => {
+    try {
+      const { awaitTask } = useTasks();
+      const { taskId } = await api.history.gatherGitcoinGrandEvents(payload);
+      const meta: TaskMeta = {
+        title: i18n
+          .t('actions.balances.gitcoin_grant.task.title', {
+            grant: 'grantId' in payload ? payload.grantId : ''
+          })
+          .toString(),
+        numericKeys: balanceKeys
+      };
+      const { result } = await awaitTask<GitcoinGrants, TaskMeta>(
+        taskId,
+        TaskType.GITCOIN_GRANT_EVENTS,
+        meta
+      );
+      return { result, message: '' };
+    } catch (e: any) {
+      return {
+        result: {},
+        message: e.message
+      };
+    }
+  };
+
+  // Reset
+  const reset = () => {
+    associatedLocations.value = [];
+    ignored.value = {};
+    useTrade().reset();
+    useAssetMovement().reset();
+    useTransaction().reset();
+    useLedgerAction().reset();
```

Comment:
small naming request here, can we make these to plural? (useTrades, useAssetMovements, useTransactions and useLedgerActions)